tasks/db_deploy/test/unit_tests/test_migrate_db.py

```diff
@@ -39,37 +39,34 @@ def tearDown(self):
     @patch("migrate_db.migrate_versions_2_to_3")
     @patch("migrate_db.migrate_versions_3_to_4")
     def test_perform_migration_happy_path(
-        self,
-        mock_migrate_v3_to_v4: MagicMock,
-        mock_migrate_v2_to_v3: MagicMock,
-        mock_migrate_v1_to_v2: MagicMock,
+            self,
+            mock_migrate_v3_to_v4: MagicMock,
+            mock_migrate_v2_to_v3: MagicMock,
+            mock_migrate_v1_to_v2: MagicMock,
     ):
         """
         Tests the perform_migration function happy paths
         """
-        for version in [1,2,3,4,5]:
+        for version in [1, 2, 3, 4, 5]:
             with self.subTest(version=version):
                 migrate_db.perform_migration(version, self.config)
 
                 # Make sure the proper migrations happens.
                 # Note that for version 2 and 3 the function is not called so
                 # overall through all the tests it is only called once.
-
-                if version == 1:
+                if version < 2:
                     mock_migrate_v1_to_v2.assert_called_once_with(self.config, False)
-                    mock_migrate_v2_to_v3.assert_called_once_with(self.config, False)
-                    mock_migrate_v3_to_v4.assert_called_once_with(self.config, True)
+                else:
+                    mock_migrate_v1_to_v2.assert_not_called()
 
-                elif version == 2:
+                if version < 3:
                     mock_migrate_v2_to_v3.assert_called_once_with(self.config, False)
-                    mock_migrate_v3_to_v4.assert_called_once_with(self.config, True)
+                else:
+                    mock_migrate_v2_to_v3.assert_not_called()
 
-                elif version == 3:
+                if version < 4:
                     mock_migrate_v3_to_v4.assert_called_once_with(self.config, True)
-
                 else:
-                    mock_migrate_v1_to_v2.assert_not_called()
-                    mock_migrate_v2_to_v3.assert_not_called()
                     mock_migrate_v3_to_v4.assert_not_called()
```

Comment:
It would work, but it requires three if/elif blocks with 3 assert_called_once/assert_not_called statements in each. n * n
This approach requires three if blocks with 2 assert_called_once/assert_not_called statements in each. n * 2 
Additionally, the elif structure requires updates to previous blocks whenever a new migration is added.

Reply:
yes that makes sense.